File: P1S3/ObjetivoVerNotas.cpp
//
// Created by danip on 06/04/2021.
//

#include <iostream>
#include "include/ObjetivoVerNotas.h"

void ObjetivoVerNotas::ejecutar(Notas& notas) {
    finalTeoria = calcularFinal(notas, TipoNota::teoria);
    finalPracticas = calcularFinal(notas, TipoNota::practicas);
    finalTotal = finalTeoria + finalPracticas;

    std::cout << "Final teoria: " << finalTeoria << std::endl;
    std::cout << "Final practicas: " << finalPracticas << std::endl;
    std::cout << "------------------------------------" << std::endl;
    std::cout << "Final total: " << finalTotal << std::endl << std::endl << std::endl;
}

double ObjetivoVerNotas::calcularFinal(Notas &notas, TipoNota tipoNota) {
    double final = 0.0;
    size_t size = notas.size();
    for(size_t i = 0; i < size; i++) {
        Nota& n = notas.getNota(i);
        if(n.getTipoNota() == tipoNota)
            final += n.getPesoFinal();
    }

    return final;
}

File: P1S3/dart/objetivo_ver_notas.dart
import 'nota.dart';
import 'notas.dart';
import 'tipo_nota.dart';

class ObjetivoVerNotas {
  Map _finalTeoria = {'notaFinal' : 0.0, 'maximaFinal' : 0.0};
  Map _finalPracticas = {'notaFinal' : 0.0, 'maximaFinal' : 0.0};
  Map _finalTotal = {'notaFinal' : 0.0, 'maximaFinal' : 0.0};

  Map _calcularFinal(Notas notas, TipoNota tipoNota) {
    double notaFinal = 0.0;
    double maximaFinal = 0.0;
    var size = notas.size;
    
    for(int i = 0; i < size; i++) {
      Nota n = notas.getNota(i);
      if(n.tipoNota == tipoNota) {
        notaFinal += n.pesoFinal;
        maximaFinal += n.maximoPesoFinal;
      }
    }
    
    return {'notaFinal' : notaFinal, 'maximaFinal' : maximaFinal};
  }

  void ejecutar(Notas notas) {
    _finalTeoria = _calcularFinal(notas, TipoNota.teoria);
    _finalPracticas = _calcularFinal(notas, TipoNota.practicas);
    _finalTotal['notaFinal'] = _finalPracticas['notaFinal'] + _finalTeoria['notaFinal'];
    _finalTotal['maximaFinal'] = _finalPracticas['maximaFinal'] + _finalTeoria['maximaFinal'];

    print("Final teoría: ${_finalTeoria['notaFinal'].toStringAsFixed(2)}"
        "/ ${(_finalTeoria['maximaFinal'] * 10).toStringAsFixed(2)} ");
    print("Final prácticas: ${_finalPracticas['notaFinal'].toStringAsFixed(2)} "
        "/ ${(_finalPracticas['maximaFinal'] * 10).toStringAsFixed(2)}");
    print("------------------------------------");
    print("Final total: ${_finalTotal['notaFinal'].toStringAsFixed(2)} "
        "/ ${(_finalTotal['maximaFinal'] * 10).toStringAsFixed(2)}\n");
  }
}